### PR TITLE
Overhaul library mapgen and add SUS bookcases

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Add SUS bookcase groups and overhaul libraries"

#### Purpose of change
Libraries don't use modern mapgen and their item spawns are randomly thrown everywhere.

#### Describe the solution
Create new SUS itemgroups and use them in libraries. Overhaul library mapgen.

#### Testing
Game loads correctly and new library spawns correctly.

#### Additional context
Testing these changes in game highlighted how we have a problem with low book variety.

### CODE REVIEW

File: data/json/mapgen_palettes/library_palette.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+[
+  {
+    "type": "palette",
+    "id": "library_palette",
+    "terrain": { "%": "t_console_broken", "p": [ [ "t_grass", 10 ], [ "t_dirt", 2 ] ], ";": "t_door_locked", "5": "t_tree_pine" },
```
Comment (I-am-Erk) [inline data/json/mapgen_palettes/library_palette.json]:
```suggestion
    "terrain": { "%": "t_console_broken", "p": "t_region_groundcover_urban", ";": "t_door_locked", "5": "t_region_tree_evergeeen" },
```

File: data/json/mapgen_palettes/library_palette.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "furniture": {
+      "c": "f_counter",
+      "d": "f_desk",
+      "[": "f_table",
+      "h": "f_chair",
+      "b": "f_stool",
+      "C": "f_sofa",
+      "B": "f_armchair",
+      "y": [ "f_indoor_plant", "f_indoor_plant_y" ],
+      "p": [ "f_datura", "f_bluebell", "f_mutpoppy", "f_dahlia", "f_flower_tulip", "f_chamomile", "f_flower_spurge", "f_lily" ],
```
Comment (I-am-Erk) [inline data/json/mapgen_palettes/library_palette.json]:
```suggestion
      "p":  "f_region_flower",
```

**Comment by Maleclypse** [comment]
When you say problem with low book variety do mean additional book snippets are needed or are you saying we need more types of books beyond the skill books, children's books, horror books...

**Comment by anothersimulacrum** [comment]
I think more types of books - to my knowledge we've got a number of books with a lot of snippets.

**Comment by Maleclypse** [comment]
> I think more types of books - to my knowledge we've got a number of books with a lot of snippets.

So when I'm looking through I don't see any snippets for children's books but sci-fi has dozens of snippets under the sci-fi book item.  Are other groups snippets outside of their item in book/misc.json?

**Comment by ymber** [comment]
Snippets for fiction would be good but I meant that we typically have 3-5 books per skill that cover the entire field with a huge number of recipes each. I'd like to split these into more books that are more specific. Before that I'd like to reverse how recipes are assigned to books so the book lists its recipes rather than every recipe listing what books it's in.

**Comment by I-am-Erk** [comment]
Vive la bibliotheque

**Pull request merged**